### PR TITLE
Bug 1236443 - [TV][2.5] App window & preview window support back navigation

### PR DESCRIPTION
**Pull request by rickychien**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=102392ac52bbe54a148c92b4f5a9949df1a31db8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=748cfbd0d02f5961371ff5a643ecc8770d5a8f0d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8963c9e64a0e4727424de197c85a5c86d4f765a8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c3d401ad02fb17e26f5703931a2654b5631b2760

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b70f47a36ea3ddbdb4d9616ab76a78f7e68f7175

File: tv_apps/smart-system/js/app_window.js [hunk 1]
```diff
@@ -1028,6 +1037,21 @@
...
 
+  AppWindow.prototype._handle_mozbrowserafterkeyup = function(evt) {
+    if (document.activeElement !== this.iframe) {
+      return;
+    }
+    if ((evt.keyCode === 27 || evt.key === 'Escape') &&
+        !evt.embeddedCancelled) {
+      var goBackReq = this.iframe.getCanGoBack();
+      goBackReq.onsuccess = () => {
+        if (goBackReq.result && !this.config.url.startsWith('app://')) {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_window.js]: !!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0364e424f368ade27ccb061cfeeb427ead046479

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4988f5035dcb0f7bc2acae0e5319671f1ceab349

File: tv_apps/smart-system/js/preview_window.js [hunk 2]
```diff
@@ -100,8 +99,18 @@
       return;
     }
     if ((evt.keyCode === 27 || evt.key === 'Escape') &&
-        !evt.embeddedCancelled) {
-      this.kill();
+        !evt.embeddedCancelled && !this.config.url.startsWith('app://')) {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/preview_window.js]: You should kill the app directly if `this.config.url.startsWith('app://')` is true.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4a8ad58f1cddbc153352aec9b810ffb7725ceb49

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=54e152c359d90c08e46381a21e12c7aca03a88d7

**Pull request merged**